### ISSUE TITLE
add a config option to make the postcss-url usage overridable

### ISSUE THREAD

**Issue by tinchoz49** [tivac/modular-css#88]

**Comment by tivac** [comment]
Context: https://github.com/tivac/modular-css/issues/35#issuecomment-187246522

**Comment by tivac** [comment]
Working on this in https://github.com/tivac/modular-css/tree/issues/88

**Comment by tivac** [comment]
@tinchoz49 will #89 cover your needs? I'll probably merge it either way since it cleans up some lingering weirdness left over from me learning PostCSS, but wanted you to look/comment first.

**Comment by tinchoz49** [comment]
Works great. Two things:

1. This is only an opinion... I don't know if "rebase" is the best name for use in this feature because is more like an option to apply a ```postcss process``` for each css file right? Now we need it for rebase our URLs but maybe there is another postcss plugin (e.g postcss-inline-svg) that needs the same functionality but is not related to rebase URLs.
I don't know what is the best name :( maybe something like ```forEachFile```.

2. Thanks for add the example of postcss-copy in your readme. To work with modular-css the example would be:
(if we keep the "rebase" name)
```js
build.plugin("modular-css/browserify", {
    // REQUIRED

    // location to write combined CSS file to
    css : "./output/css/here.css",

    // OPTIONAL

    // output JSON file containing all composes/scoped class names
    json : "./output/classes.json",

    // PostCSS plugins to run before/after processing
    before : [ require("postcss-import") ],
    after  : [ require("cssnano")() ],

    // PostCSS plugins to use instead of postcss-url for relative url() rebasing
    rebase : [ require('postcss-copy')({
            src: './',
            dest: 'output/css',
            relativePath(dirname, fileMeta, result, opts) {
                return opts.dest;
            }
        }) 
   ],

    // Source maps in the output
    map : true
});
```

**Comment by tivac** [comment]
1. I originally kept it as `rebase` because that's what it does, but I agree that it's weird. After thinking about it more I like `each`. `before`, `each`, `after` feels pretty good.
2. I'm definitely not going to put a big chunk of documentation into the readme for a plugin, I'll probably just invent a plugin name and leave figuring out configuration to the end-user (since showing how to use a specific plugin correctly isn't what the modular-css readme is for)

**Comment by tinchoz49** [comment]
:+1: both decisions! I agree in use ```each```

**Comment by tivac** [comment]
Sorry, went to add `each` and didn't love it. Went with the following:

`before` - applied before files are processed
`after` - applied after files are processed
`done` - applied once files are combined

**Comment by tivac** [comment]
`modular-css@0.12.0` is published with the changes in #89, hope it helps!
